src/garage/torch/algos/vpg.py

```diff
@@ -135,38 +139,65 @@ def train_once(self, itr, paths):
         obs, actions, rewards, valids, baselines = self.process_samples(
             itr, paths)
 
-        loss = self._compute_loss(itr, obs, actions, rewards, valids,
-                                  baselines)
-
-        self._old_policy.load_state_dict(self.policy.state_dict())
-
-        self._optimizer.zero_grad()
-        loss.backward()
+        with torch.no_grad():
+            loss_before = self._compute_loss(itr, obs, actions, rewards,
+                                             valids, baselines)
+            kl_before = self._compute_kl_constraint(obs)
+
+        step_size = self.minibatch_size if self.minibatch_size else len(paths)
+        for epoch in range(self.optim_epochs):
+            shuffled_ids = torch.randperm(len(paths))
```

Comment:
is there a reason you can't use https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.random.permutation.html#numpy.random.permutation instead?

Reply:
These is no reason.
I just want to use PyTorch API (instead of numpy) because most of the code use PytTorch libraries.

Reply:
this is kind of tortured and suggests to the user that something with GPUs is going on (which it isn't). please just use numpy when possible.

Reply:
fixed